pkg/api/metadata.go

```diff
@@ -143,7 +143,11 @@ func BuildCacheFor(metadata Metadata) ImageStreamTagReference {
 }
 
 func ImageVersionLabel(fromTag PipelineImageStreamTagReference) string {
-	return fmt.Sprintf("io.openshift.ci.from.%s", fromTag)
+	label := fmt.Sprintf("io.openshift.ci.from.%s", fromTag)
+	if len(label) > 63 { // The max length of a label is 63 chars
```

Comment:
NIT: Reuse this? (not exactly the same but pretty close)

https://github.com/openshift/ci-tools/blob/70247925de695c48b3f5af619d295394d079c2df/pkg/steps/utils/labels.go#L59

Reply:
I had to move the entire file to negotiate a circular import dependency, but I think this is a better spot for it anyways.